### PR TITLE
don't using kaldi reco2dur  and fix some error in bin/lhotse

### LINKED ISSUE #320: error of cut_set.compute_and_store_features
When I use snowfall to run an asr model, at the prepare Lhotse formate data,I encountered the following problems.
Error detail log is as follows:
```
# python3 ./prepare_seame.py 
# Invoked at Sat Jun 19 00:39:48 +08 2021 from node09
#
# Started at Sat Jun 19 00:39:48 +08 2021 on node09
WARNING:root:There are 15 recordings that do not have any corresponding supervisions in the SupervisionSet.
Dataset parts:   0%|          | 0/4 [00:00<?, ?it/s]Parts we will prepare:  ('dev_man', 'dev_sge', 'train_trn', 'train_cv')
Musan manifest preparation:
Feature extraction:
partition is dev_man
Processing dev_man
Extracting and storing features (chunks progress):  45%|████▌     | 9/20 [04:55<06:00, 32.80s/it]
Dataset parts:   0%|          | 0/4 [04:56<?, ?it/s]15%|█▌        | 3/20 [04:55<22:49, 80.57s/it] 
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home3/md510/anaconda3/envs/foo_k2/lib/python3.8/concurrent/futures/process.py", line 239, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/cut.py", line 2064, in compute_and_store_features
    return CutSet.from_cuts(
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/cut.py", line 1407, in from_cuts
    return CutSet(cuts=index_by_id_and_check(cuts))
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/utils.py", line 316, in index_by_id_and_check
    for m in manifests:
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/cut.py", line 2066, in <genexpr>
    cut.compute_and_store_features(
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/cut.py", line 424, in compute_and_store_features
    samples=self.load_audio(),
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/cut.py", line 394, in load_audio
    return self.recording.load_audio(
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/audio.py", line 247, in load_audio
    audio = assert_and_maybe_fix_num_samples(
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/audio.py", line 823, in assert_and_maybe_fix_num_samples
    raise ValueError("The number of declared samples in the recording diverged from the one obtained "
ValueError: The number of declared samples in the recording diverged from the one obtained when loading audio (offset=0, duration=1199.328). This could be internal Lhotse's error or a faulty transform implementation. Please report this issue in Lhotse and show the following: diff=9, audio.shape=(1, 19189239), recording=Recording(id='ui03faz_0101', sources=[AudioSource(type='file', channels=[0], source='/home4/md510/w2018/original_seame/wavdata/interview/ui03faz_0101/ui03faz_0101.wav')], sampling_rate=16000, num_samples=19189248, duration=1199.328, transforms=None)
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./prepare_seame.py", line 167, in <module>
    main()
  File "./prepare_seame.py", line 135, in main
    cut_set = cut_set.compute_and_store_features(
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/cut.py", line 2118, in compute_and_store_features
    cuts_with_feats = combine(progress(f.result() for f in futures))
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/manipulation.py", line 27, in combine
    return reduce(add, manifests)
  File "/home3/md510/anaconda3/envs/foo_k2/lib/python3.8/site-packages/tqdm/std.py", line 1178, in __iter__
    for obj in iterable:
  File "/home3/md510/w2020/k2_fsa_2021/lhotse/lhotse/cut.py", line 2118, in <genexpr>
    cuts_with_feats = combine(progress(f.result() for f in futures))
  File "/home3/md510/anaconda3/envs/foo_k2/lib/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/home3/md510/anaconda3/envs/foo_k2/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
ValueError: The number of declared samples in the recording diverged from the one obtained when loading audio (offset=0, duration=1199.328). This could be internal Lhotse's error or a faulty transform implementation. Please report this issue in Lhotse and show the following: diff=9, audio.shape=(1, 19189239), recording=Recording(id='ui03faz_0101', sources=[AudioSource(type='file', channels=[0], source='/home4/md510/w2018/original_seame/wavdata/interview/ui03faz_0101/ui03faz_0101.wav')], sampling_rate=16000, num_samples=19189248, duration=1199.328, transforms=None)
# Ended (code 256) at Sat Jun 19 00:44:59 +08 2021, elapsed time 311 seconds
```
Could you help me to solve it? @pzelasko 
 Thanks a lot.

### PR DESCRIPTION
**Pull request by shanguanma**
closed #320 
I add some code to ignore read duration from reco2dur and read real audio_path in` wav.scp` to get duration.
Test time consumption: about 100 hours and ten seconds.

### CODE REVIEW

File: lhotse/kaldi.py [hunk 1]
```diff
@@ -7,6 +7,19 @@
 from lhotse.audio import AudioSource, Recording, RecordingSet
 from lhotse.supervision import SupervisionSegment, SupervisionSet
 from lhotse.utils import Pathlike, is_module_available
+from lhotse.audio import _audioread_info 
+
+def from_file(
+    path: Pathlike,
+):
```
Comment (csukuangfj) [inline lhotse/kaldi.py:14]: Would you mind adding documentation and return type hint to this function?

File: lhotse/kaldi.py [hunk 2]
```diff
@@ -7,6 +7,19 @@
 from lhotse.audio import AudioSource, Recording, RecordingSet
 from lhotse.supervision import SupervisionSegment, SupervisionSet
 from lhotse.utils import Pathlike, is_module_available
+from lhotse.audio import _audioread_info
```
Comment (csukuangfj) [inline lhotse/kaldi.py:10]: Shall we rename `_audioread_info` to `audioread_info`
as it is used outside `audio.py`? @pzelasko

File: lhotse/kaldi.py (hunk 1 above)
Comment (shanguanma) [reply lhotse/kaldi.py:14]: ok,  I will add it.

File: lhotse/kaldi.py (hunk 2 above)
Comment (shanguanma) [reply lhotse/kaldi.py:10]: I agree with @csukuangfj 's suggestion.

File: lhotse/kaldi.py [hunk 3]
```diff
@@ -1,12 +1,32 @@
...
+from typing import Any, Dict, Optional, Tuple, NamedTuple
 
 from lhotse import CutSet, FeatureSet, Features, Seconds
 from lhotse.audio import AudioSource, Recording, RecordingSet
 from lhotse.supervision import SupervisionSegment, SupervisionSet
 from lhotse.utils import Pathlike, is_module_available
+from lhotse.audio import audioread_info 
+
+
+def from_file(
```
Comment (csukuangfj) [inline lhotse/kaldi.py:13]: This function is to get the duration of a sound file.
`get_duration` maybe is a better name than `from_file`.

File: lhotse/kaldi.py [hunk 4]
```diff
@@ -28,13 +48,10 @@ def load_kaldi_data_dir(
...
-    reco2dur = path / 'reco2dur'
-    if not reco2dur.is_file():
-        raise ValueError(f"No such file: '{reco2dur}' -- fix it by running: utils/data/get_reco2dur.sh <data-dir>")
-    with reco2dur.open() as f:
-        for line in f:
-            recording_id, dur = line.strip().split()
-            durations[recording_id] = float(dur)
+    for recording_id, path_or_cmd in recordings.items():
+        duration = from_file(path_or_cmd)
+        #duration = info.duration
```
Comment (csukuangfj) [inline lhotse/kaldi.py:53]:
```suggestion
```

File: lhotse/bin/lhotse [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+#!/usr/bin/env python3
 """
 Use this script like:
 
 $ lhotse --help
 $ lhotse make-feats --help
 $ lhotse make-feats --compressed recording_manifest.yml mfcc_dir/
 $ lhotse write-default-feature-config feat-conf.yml
-$ lhotse convert-kaldi data/train/ 16000 train_manifests/
+$ lhotse kaldi import  data/train 16000 train_manifests/
```
Comment (csukuangfj) [inline lhotse/bin/lhotse:9]: Nit: There is an extra space after `import`.

File: lhotse/kaldi.py [hunk 6]
```diff
@@ -1,12 +1,32 @@
 import warnings
 from collections import defaultdict
 from pathlib import Path
-from typing import Any, Dict, Optional, Tuple
+from typing import Any, Dict, Optional, Tuple, NamedTuple
```
Comment (csukuangfj) [inline lhotse/kaldi.py:4]: Nit: The imported things are usually sorted alphabetically.
```suggestion
from typing import Any, Dict, NamedTuple, Optional, Tuple
```

File: lhotse/kaldi.py [hunk 7]
```diff
@@ -1,12 +1,32 @@
...
 from lhotse import CutSet, FeatureSet, Features, Seconds
 from lhotse.audio import AudioSource, Recording, RecordingSet
 from lhotse.supervision import SupervisionSegment, SupervisionSet
 from lhotse.utils import Pathlike, is_module_available
+from lhotse.audio import audioread_info 
+
+
+def from_file(
+    path: Pathlike,
+) -> Tuple[float]:
```
Comment (csukuangfj) [inline lhotse/kaldi.py:15]:
```suggestion
) -> float: 
```

File: lhotse/kaldi.py [hunk 8]
```diff
@@ -47,7 +64,7 @@ def load_kaldi_data_dir(
                 )
             ],
             sampling_rate=sampling_rate,
-            num_samples=int(durations[recording_id] * sampling_rate),
+            num_samples=int(durations[recording_id]* sampling_rate),
```
Comment (csukuangfj) [inline lhotse/kaldi.py:67]: Nit: There should be a space before `*`.

Comment (shanguanma) [reply lhotse/kaldi.py]: Done!

File: lhotse/kaldi.py (hunk 7 above)
Comment (shanguanma) [reply lhotse/kaldi.py]: Done!

File: lhotse/kaldi.py (hunk 6 above)
Comment (shanguanma) [reply lhotse/kaldi.py]: I will remove NamedTuple, because it isn't used.

File: lhotse/bin/lhotse (hunk 5 above)
Comment (shanguanma) [reply lhotse/bin/lhotse]: Done!

File: lhotse/kaldi.py (hunk 3 above)
Comment (shanguanma) [reply lhotse/kaldi.py]: I agree with you. Done!

File: lhotse/kaldi.py (hunk 4 above)
Comment (shanguanma) [reply lhotse/kaldi.py]: Done!

**Comment by pzelasko** [comment]
Looks good to me -- one thing that I think could be improved is that for large datasets, this will be very slow. We could add a `num_jobs` argument to the `load_kaldi_data_dir` function and use a `ProcessPoolExecutor` to parallelize the duration computation. But this can easily be a follow-up PR, so I'll merge this as it is.

**Pull request merged**